site/content/concepts/tenancy.md

```diff
@@ -1,12 +1,149 @@
+Such an isolation is essential for enabling a scalable distributed architecture to handle independent subsets as if each subset had its
+own installation (which would be much harder to maintain and would not benefit from runtime cost sharing).
 
-{{% warning %}}
-This concept is currently part of active development and discussion.
-Please join our mailing list or consult the IRC channel in case you have any questions or want to participate.
-{{% /warning %}}
+This is why Hono implements a multitenancy concept.
+
+## Hono's multitenancy concept
+
+Hono's multitenancy concept is based on handling tenants as own *entities*. All functionality of Hono is 
+provided in the context of a previously created tenant - except the creation of a tenant itself. 
+
+
+In the following the different aspects of multitenancy in Hono is addressed and a full overview of the concept is given.
+
+
+## The Tenant API
+
+By means of the [Tenant API]({{< relref "Tenant-API.md" >}}) Hono handles tenants as own *entities*.
+
+The API defines how to *retrieve* the details of a specific tenant. This offers the possibility to handle arbitrary
+properties on the level of a tenant (see e.g. [Protocol adapter configuration]({{< relref "#protocol-adapter-configuration" >}})).
+
+For convenience, there are CRUD operations for the handling of tenants, which can be found in the 
+[Device Registry]({{< relref "device-registry.md#managing-tenants" >}}).
+
+## Protocol Adapters respect the Tenant API
+
+When a device connects to one of Hono's protocol adapters, the adapter determines the tenant this device belongs to.
+How this is done is described for the protocol adapters in the User Guide.
+
+After the tenant is determined, the adapter retrieves the details of the determined tenant by means of the Tenant API.
+Only if the tenant exists and is enabled the adapter further processes the data of the device that is connecting. Otherwise
+the connection will be closed.
+
+## Protocol Adapter Configuration
+
+Protocol adapters retrieve parts of their configuration on a tenant level by using the details of the tenant that was determined.
+
+This includes e.g. if a specific protocol adapter is enabled at all for this tenant, allowing to define tenants with 
+only a subset of Hono's functionality. This feature is foreseen to be especially important for production setups.
+
+An example is to define a tenant that is only enabled for MQTT access, but is not allowed to connect to the HTTP adapter.
+
+Please refer to the [Tenant API]({{< relref "Tenant-API.md" >}}) to find out which protocol adapter properties are
+configurable on a tenant level. The number of such properties may grow during further development of Hono.
+
+## AMQP 1.0 endpoints
+
+The AMQP 1.0 endpoints for all APIs of Hono are scoped to a tenant, by using the scheme `<api>/TENANT/...`.
+
+Examples are:
+
+- `telemetry/TENANT/device`
```

Comment:
the APIs are usually not scoped to a particular device, IMHO we should therefore remove the trailing `/device` segments ...